assets/src/hooks/useDetour.ts

```diff
@@ -27,9 +27,29 @@ export const useDetour = (input: UseDetourInput) => {
 
   // Record snapshots when changed
   useEffect(() => {
-    const snapshotSubscription = actorRef.subscribe(() => {
-      const serializedSnapshot = JSON.stringify(actorRef.getPersistedSnapshot())
+    const snapshotSubscription = actorRef.subscribe((snap) => {
+      const persistedSnapshot = actorRef.getPersistedSnapshot()
+      const serializedSnapshot = JSON.stringify(persistedSnapshot)
       localStorage.setItem("snapshot", serializedSnapshot)
+
+      // check for no-save tag before
+      if (snap.hasTag("no-save")) {
+        return
+      }
+
+      actorRef.getSnapshot().can({ type: "detour.save.begin-save" }) &&
```

Comment:
What happens if `.send` is called even if `.can` returns false? I assumed it wouldn't do anything, but I may have made that up.

Reply:
🤔 The `can` clause serves as a guard here, so the `send` cannot execute if the `can` returns false, right? Now I'm second guessing everything

Reply:
Well, I tried removing the `.can` check and it caused the tests to hang, so I think my guess about how `.send` works was wrong 😅

Reply:
if `.send` is called, even though it _can't_ be handled more than once(with our current machine), it seems to cause the `subscribe` callback to be called again with a new snapshot, which gets into a loop where it's constantly triggering `send`->`subscribe`, unfortunately you'd need to guard the `send` some other way because of that, but it is a little surprising that `.send` doesn't do a `.can` internally and prevent `subscribe` from being invoked (though maybe it makes sense because `subscribe` does not describe if you're subscribing to anything _other_ than just snapshot updates and what causes a snapshot update does not seem to be explicitly defined either.)

Reply:
Ohhh oh oh, I understand your initial question now, after Kayla's reply. Yeah, we did that exact thing and tried just `send`ing, but ran into issues